pkg/util/miseadapter/miseadapter.go

```diff
@@ -0,0 +1,194 @@
+		// OriginalUri is the Uri of the original request being validated.
+		OriginalUri string
+
+		// OriginalMethod is the method of the original request being validated.
+		OriginalMethod string
+
+		// OriginalIpAddress is the IP address of original request being validated.
+		OriginalIPAddress string
+
+		// AuthorizationHeader is the authorization header of the original request being validated.
+		AuthorizationHeader string
+
+		// ReturnAllActorClaims specifies whether to return all claims from the actor token.
+		ReturnAllActorClaims bool
+
+		// ReturnAllSubjectClaims specifies whether to return all claims from the subject token.
+		ReturnAllSubjectClaims bool
+
+		// ActorClaimsToReturn specifies the specific claims to return from the actor token if present.
+		ActorClaimsToReturn []string
+
+		// SubjectClaimsToReturn specifies the specific claims to return from the subject token.
+		SubjectClaimsToReturn []string
+	}
+
+	// Result is the authentication result.
+	Result struct {
+		// ActorClaims is the set of claims extracted from the actor token based on the input options.
+		ActorClaims map[string][]string
+
+		// SubjectClaims is the set of claims extracted from the subject token based on the input options.
+		SubjectClaims map[string][]string
+
+		// ErrorDescription is the description of the error from validating the token.
+		ErrorDescription []string
+
+		// WWWAuthenticate is the value of the WWWAuthenticate header when the request is unauthorized.
+		WWWAuthenticate []string
+
+		// StatusCode is the status code that the Mise container returns as a result of validating the token.
+		StatusCode int
+	}
+)
+
+// New creates a Client able to delegate token validation.
+func New(httpClient *http.Client, miseAddress string) *Client {
+	return &Client{
+		httpClient:  httpClient,
+		miseAddress: miseAddress,
+	}
+}
+
+// ValidateRequest transforms the Input object to a request to the Mise container and returns
+// the Result object.
+func (c Client) ValidateRequest(ctx context.Context, input Input, log *logrus.Entry) (Result, error) {
+	req, reqErr := createRequest(ctx, c.miseAddress, input)
+	if reqErr != nil {
+		return Result{}, reqErr
+	}
+
+	resp, respErr := c.httpClient.Do(req)
+	if respErr != nil {
+		return Result{}, respErr
+	}
+
+	defer resp.Body.Close()
+
+	return parseResponseIntoResult(resp)
+}
+
+func createRequest(ctx context.Context, miseAddress string, input Input) (*http.Request, error) {
+	req, reqErr := http.NewRequestWithContext(ctx, http.MethodPost, miseAddress+"/ValidateRequest", nil)
+	if reqErr != nil {
+		return nil, reqErr
+	}
+
+	req.Header.Add("Authorization", input.AuthorizationHeader)
+	req.Header.Add("Original-Uri", input.OriginalUri)
+	req.Header.Add("Original-Method", input.OriginalMethod)
+	req.Header.Add("X-Forwarded-For", input.OriginalIPAddress)
+
+	if input.ReturnAllActorClaims {
+		req.Header.Add("Return-All-Actor-Token-Claims", "1")
+	} else {
+		for _, val := range input.ActorClaimsToReturn {
```

Comment:
I didn't see this being used in this PR in any code but tests. Is that a miss, do we intend on adding it later (if so, which ticket?), or do we only include this here for compliance to a spec?

Reply:
Currently, our primary concern is determining whether a request is authenticated. Therefore, we are not including the header in the `input` or passing any actor claims in the request to MISE. Retrieving all actor claims would be inefficient, as it would result in unnecessarily large responses, impacting performance.  

Additionally, since granular authorization based on client roles or other claims is not a requirement at this stage, we are not fetching specific actor claims either. This code block serves as a placeholder for potential future implementations, should the need for more detailed authorization arise in later phases.

Reply:
Ok I see that returning all would be good and like that the code as it is now seeks to return an empty set of claims for efficiency.

Reply:
With regards to "This code block serves as a placeholder for potential future implementations, should the need for more detailed authorization arise in later phases.", isn't that code currently unecessary? I think it is better to put that in when we actually do need it? We have be burned in the past where we have tried to future-proof and then it never got implemented (or the strategy changed slightly) and we were left with tech debt which we needed to clean-up. WDYT @geowa4 ?

Reply:
Yeah, I appreciate the use of `[]string` to return an empty set of claims (hooray efficiency), but if there's a large time gap between now and when this block is used, the API could have changed and we will have a bug to find.

Reply:
I see we can work on this by creating a JIRA ticket and mentioning about either utilizing this section of code in next phase while we Enforce MISE or get rid of it completely. Getting this code out includes removing it references i other files where this is being parsed from the response received from MISE. 
This way we have a record so this granular access control section doesn't get missed while we also make sure it is removed if not used later !
@geowa4 @brizrobbo WDYT ?

Reply:
+1 If we have an impending action item I can get behind deferring the decision to keep/remove. For the future, keeping our PRs to only the minimally required paths reduces complexity, increases readability, and minimizes our surface area to attack, bugs, and generally getting stale.